### PR TITLE
PR from fork/331

### PR DESCRIPTION
**Pull request by MichaelMcKinsey1**
PR from #331 Integrate Caliper

### CODE REVIEW

**Comment by rhornung67** [comment]
@MichaelMcKinsey1 I merged develop into this and resolved the merge conflicts. When checks pass, is this ready to merge?

**Comment by MichaelMcKinsey1** [comment]
> @MichaelMcKinsey1 I merged develop into this and resolved the merge conflicts. When checks pass, is this ready to merge?

1 more issue came up today. Once it is addressed we should be good to go.

**Comment by MichaelMcKinsey1** [comment]
> > @MichaelMcKinsey1 I merged develop into this and resolved the merge conflicts. When checks pass, is this ready to merge?
> 
> 1 more issue came up today. Once it is addressed we should be good to go.

Done in [51144fb](https://github.com/LLNL/RAJAPerf/pull/352/commits/51144fb282f31aac4f02f0970f6eb3470dc2d07e).  

Also, @daboehme wanted to cleanup the caliper implementation before the merge, so that will come soon.

**Comment by rhornung67** [comment]
Sounds good. I review and get some others to also review when you're ready. Please let me know.

**Comment by MichaelMcKinsey1** [comment]
> > > @MichaelMcKinsey1 I merged develop into this and resolved the merge conflicts. When checks pass, is this ready to merge?
> > 
> > 
> > 1 more issue came up today. Once it is addressed we should be good to go.
> 
> Done in [51144fb](https://github.com/LLNL/RAJAPerf/pull/352/commits/51144fb282f31aac4f02f0970f6eb3470dc2d07e).
> 
> Also, @daboehme wanted to cleanup the caliper implementation before the merge, so that will come soon.

#359

**Comment by MichaelMcKinsey1** [comment]
@rhornung67 Changes merged. Ready now.

File: src/algorithm/MEMCPY-Cuda.cpp
Comment (rhornung67) [inline src/algorithm/MEMCPY-Cuda.cpp:160]: @MrBurmark please take a look at these changes and indicate whether you have any questions or concerns.

**Review by rhornung67 — COMMENTED** [review]
A couple of minor comments.

File: scripts/lc-builds/blueos_nvcc_clang_caliper.sh
Comment (MrBurmark) [inline scripts/lc-builds/blueos_nvcc_clang_caliper.sh:54]: Should the script take a path to these libraries to make them more general?

File: src/algorithm/MEMCPY-Cuda.cpp
Comment (MrBurmark) [reply src/algorithm/MEMCPY-Cuda.cpp:160]: This particular change is to allow capture of the block size as metadata. I'm not thrilled with this but I'm not sure there is a better way to do it either. I have some ideas about refactoring tunings that may improve things a bit but that's longer term.

File: src/common/Executor.cpp
Comment (MrBurmark) [inline src/common/Executor.cpp:203]: Use find instead of try catch.

File: src/common/Executor.cpp
Comment (MrBurmark) [inline src/common/Executor.cpp:781]:
```suggestion
            std::string const& tstr = tuning_name_idx_pair.first;
```

File: src/algorithm/MEMCPY-Cuda.cpp
Comment (MichaelMcKinsey1) [reply src/algorithm/MEMCPY-Cuda.cpp:160]: Most kernels' block size are being set by a change in `src/common/GPUUtils.hpp::RAJAPERF_GPU_BLOCK_SIZE_TUNING_DEFINE_BOILERPLATE`. But some like `MEMCPY-Cuda.cpp` don't contain the boilerplate so I had to put them in manually like above. To clarify this was to get per kernel block size and not for the run metadata.

File: scripts/lc-builds/blueos_nvcc_clang_caliper.sh
Comment (MichaelMcKinsey1) [reply scripts/lc-builds/blueos_nvcc_clang_caliper.sh]: [011643d](https://github.com/LLNL/RAJAPerf/pull/352/commits/011643dce385978d67131478e8b664bbb4d0c5bd)

File: src/common/Executor.cpp
Comment (MichaelMcKinsey1) [reply src/common/Executor.cpp]: [962334b](https://github.com/LLNL/RAJAPerf/pull/352/commits/962334b53464aa150e3a9ac9c44343071430943c)

**Review by rhornung67 — APPROVED** [review]

**Review by MrBurmark — APPROVED** [review]

**Comment by rhornung67** [comment]
@MichaelMcKinsey1 I merged develop into this. There were several merge conflicts in Executor.cpp that I resolved because I refactored Executor::setupSuite(). When you have time, please take a look to make sure I didn't screw up anything you did. When you are OK with it, I will merge it. Thank you.

**Comment by rhornung67** [comment]
No worries.

**Comment by MichaelMcKinsey1** [comment]
@rhornung67 Fixed and re-merged develop.

**Comment by rhornung67** [comment]
> @rhornung67 Fixed and re-merged develop.

Thanks for looking over @MichaelMcKinsey1 I will merge this when checks come back green. It may be a couple of days since our ruby platform is down sue to cooling issues.

**Pull request merged**
